File: FizzBuzz/java/src/main/java/FiltersProvider.java
import java.util.List;
import java.util.ArrayList;

enum FiltersProvider {
    $;
    private static final List<Filter> filtererList = new ArrayList<Filter>();

    static void add(final Filter filterer) { filtererList.add(filterer); }

    static void removeAll() { filtererList.clear(); }

    static List<Filter> all() {
	return new ArrayList(filtererList);
    }
}

File: FizzBuzz/TDD/JavaScript/src/FiltersProvider.js
const filters = new Set()

class FiltersProvider {
    add(filter) {
	filters.add(filter)
	return filters.size
    }

    removeAll() {
	 filters.clear()
	 return filters.size
    }
    
    all() {
	return new Set(Array.from(filters))
    }
}

module.exports = { filtersProvider: new FiltersProvider() }
